### PR TITLE
Decouple repository project

### PR DESCRIPTION
**Pull request by rafamanzo**

### CODE REVIEW

**Comment by danielkza** [comment]
I think this may be missing a couple of commits with tests, I'll add them.

**Comment by rafamanzo** [comment]
There was a broken acceptance test that I hope to fix with my last commit.

You are referring to this or other tests are missing?

**Comment by danielkza** [comment]
We wrote some unit tests for `ProjectAttributes`.

**Comment by diegoamc** [comment]
Seems ok to me. Good work! :)

File: config/routes.rb [hunk 1]
```diff
@@ -4,14 +4,16 @@
...
-      resources :repositories, except: [:update, :index]
-      get '/repositories/:id/modules/:module_result_id' => 'repositories#show', as: :repository_module
-      post '/repositories/:id/state' => 'repositories#state', as: :repository_state
-      post '/repositories/:id/state_with_date' => 'repositories#state_with_date', as: :repository_state_with_date
-      put '/repositories/:id' => 'repositories#update', as: :repository_update
-      get '/repositories/:id/process' => 'repositories#process_repository', as: :repository_process
+      resources :repositories, only: [:new, :create]
     end
 
+    resources :repositories, except: [:update, :index]
```
Comment (pedroscocco) [inline config/routes.rb]: I think we are going to need an Index for repositories now that they can be created without a project. Or else they wont be accessible from anywhere in the website.

File: spec/controllers/repositories_controller_spec.rb [hunk 2]
```diff
@@ -136,7 +136,7 @@
           delete :destroy, id: repository.id, project_id: project.id.to_s
         end
 
-        it { is_expected.to redirect_to(project_path(repository.project_id)) }
+        it { is_expected.to redirect_to(projects_path) }
```
Comment (pedroscocco) [inline spec/controllers/repositories_controller_spec.rb]: Maybe this should redirect to Repository#index. Whenever it exists

File: config/routes.rb (hunk 1 above)
Comment (rafamanzo) [inline config/routes.rb]: Soon but not yet. The objective here is just to decouple repository from project logically. The interface is still coupled and is the objective of a next story.

File: spec/controllers/repositories_controller_spec.rb (hunk 2 above)
Comment (rafamanzo) [inline spec/controllers/repositories_controller_spec.rb]: This view will exist just in a future PR to be done.

**Pull request merged**